packages/react-core/src/components/Label/Label.tsx

```diff
@@ -1,16 +1,79 @@
 import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Label/label';
+import { Button } from '../Button';
 import { css } from '@patternfly/react-styles';
+import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';
 
 export interface LabelProps extends React.HTMLProps<HTMLSpanElement> {
   /** Content rendered inside the label. */
   children: React.ReactNode;
   /** Additional classes added to the label. */
   className?: string;
+  /** Color of the label. */
+  color?: 'blue' | 'cyan' | 'green' | 'orange' | 'purple' | 'red' | 'grey';
+  /** Variant of the label. */
+  variant?: 'outline' | 'filled';
+  /** Icon added to the left of the label text. */
+  icon?: React.ReactNode;
+  /** Close click callback for removable labels. If present, label will have a close button. */
+  onClose?: (event: React.MouseEvent) => void;
+  /** Aria label of the close button. */
+  closeBtnAriaLabel?: string;
+  /** Id of the close button. */
+  closeBtnId?: string;
```

Comment:
I already know someone is going to want to customize the button they put in here... Can we just make the close button a `React.ReactNode` that the consumer can pass in, but has a default of:
```jsx
    <Button
      type="button"
      variant="plain"
      onClick={onClose}
      aria-label={'some-default'}
      id={'some-default'}
      aria-labelledby={`'some-default'`}
    >
      <TimesIcon />
    </Button>
```
if they pass `onClose`? It also cleans up having to support the `closeBtnAriaLabel`, `closeBtnId`, and `closeBtnTextId` props.

Reply:
I'll add a node property for a full custom button, and will condense the various default button props into a general 'closeBtnProps' object. That way users can pass in whatever labelling/id they want without needing multiple props or to define a full button.

Reply:
Works for me!